### ISSUE TITLE
Subnetwork iterate Components just yields all components

### ISSUE THREAD

**Issue by Cellophil** [PyPSA/PyPSA#380]
Using pypsa 0.19.0, I'm investigating a seemingly spurious shift in the index of the ptdf calculation of the subnetwork module (subnetwork PTDF yields a numpy matrix of correct size, so this may well be my code's fault). While doing so I stumbled over this, posting a minimal code here:

```
import pandas as pd
import numpy as np
import pypsa

n = pypsa.Network()
n.add('Bus', 'Bus0-1')
n.add('Bus', 'Bus0-2')

n.add('Bus', 'Bus1-1')
n.add('Bus', 'Bus1-2')

n.add('Line', 'line0', bus0='Bus0-1',bus1='Bus0-2', x=1)
n.add('Line', 'line1', bus0='Bus1-1',bus1='Bus1-2', x=1)

n.determine_network_topology()
print(n.sub_networks)
s = pypsa.SubNetwork(n, '0')

print(next(s.iterate_components({'Line'})).df)
```

The network decays into two Subnetworks as it should be, but iterate components yields just all lines. There doesn't seem to be a filter inplace in the code. This is presumably a bug, or at best a misleading function term? Subnetwork probably should not see any components outside it's realm. Iterate components is used in various places, so maybe I'm wrong and this is the expected behavior. In that case, please let me know. 

Apart from this, I would love to get a pandas object for the ptdf back - it would just be very convenient having the line and buses indices in place. I understand that it's more complicated with transformers as additional passive components, probably making a multiindex necessary, but nevertheless.

**Comment by Cellophil** [comment]
I finally found the error, s.PTDF is indexed by an ordered version of buses, which is accessible via s.buses_o. This is easy to miss I think - adding the index to the PTDF would definitely help.

**Comment by fneum** [comment]
Yes, I agree. Turning PTDF (and also other matrices like BODF) into pd.DataFrame with proper axis labels would be good.
